From 4d4d6eb6e8b519fec7ca5b689762bb964a7ce186 Mon Sep 17 00:00:00 2001
From: Fangrui Song <i@maskray.me>
Date: Fri, 31 May 2024 20:25:55 -0700
Subject: [PATCH] [ELF] findOrphanPos: avoid redundant getRankProximity call.
 NFC

---
 lld/ELF/Writer.cpp | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/lld/ELF/Writer.cpp b/lld/ELF/Writer.cpp
index c2ccc4f49ad2..05f2e95def4b 100644
--- a/lld/ELF/Writer.cpp
+++ b/lld/ELF/Writer.cpp
@@ -936,17 +936,22 @@ findOrphanPos(SmallVectorImpl<SectionCommand *>::iterator b,
   }
 
   // Find the first element that has as close a rank as possible.
-  auto i = std::max_element(b, e, [=](SectionCommand *a, SectionCommand *b) {
-    return getRankProximity(sec, a) < getRankProximity(sec, b);
-  });
-  if (i == e)
+  if (b == e)
     return e;
+  int proximity = getRankProximity(sec, *b);
+  auto i = b;
+  for (auto j = b; ++j != e;) {
+    int p = getRankProximity(sec, *j);
+    if (p > proximity) {
+      proximity = p;
+      i = j;
+    }
+  }
   if (!isa<OutputDesc>(*i))
     return e;
   auto foundSec = &cast<OutputDesc>(*i)->osec;
 
   // Consider all existing sections with the same proximity.
-  int proximity = getRankProximity(sec, *i);
   unsigned sortRank = sec->sortRank;
   if (script->hasPhdrsCommands() || !script->memoryRegions.empty())
     // Prevent the orphan section to be placed before the found section. If
-- 
2.47.0.163.g1226f6d8fa-goog

